### ISSUE TITLE
Strange dependency / build problems with cabal

### ISSUE THREAD

**Issue by yaxu** [tidalcycles/Tidal#525]
Cabal is failing for many people. This looks typical: https://toplap.lurk.org/t/i-cant-install-tidal/607/3

Even though the matrix looks fine: https://matrix.hackage.haskell.org/#/package/tidal
as do the travis tests: https://travis-ci.org/tidalcycles/Tidal

**Comment by yaxu** [comment]
Any tips @hvr?

**Comment by hvr** [comment]
@yazu, I had a very brief look at this, and I can say the `MonadFail` compile error is legit; it can easily be reproduced by e.g.

```
$ cabal v2-build -w ghc-8.2.2 --constraint 'parsec == 3.1.11'
Build profile: -w ghc-8.2.2 -O1
In order, the following will be built (use -v for more details):
 - tidal-1.0.13 (lib) (first run)
Preprocessing library for tidal-1.0.13..
Building library for tidal-1.0.13..
[16 of 22] Compiling Sound.Tidal.ParseBP ( src/Sound/Tidal/ParseBP.hs, /tmp/tidal/dist-newstyle/build/x86_64-linux/ghc-8.2.2/tidal-1.0.13/build/Sound/Tidal/ParseBP.o ) [Text.ParserCombinators.Parsec.Token changed]

src/Sound/Tidal/ParseBP.hs:401:24: error:
    • No instance for (Control.Monad.Fail.MonadFail
                         (Text.Parsec.Prim.ParsecT String () Identity))
        arising from a use of ‘readColourName’
    • In the first argument of ‘(<?>)’, namely ‘readColourName name’
      In a stmt of a 'do' block:
        colour <- readColourName name <?> "known colour"
      In the expression:
        do name <- many1 letter <?> "colour name"
           colour <- readColourName name <?> "known colour"
           return $ TPat_Atom colour
    |
401 |              colour <- readColourName name <?> "known colour"
    |                        ^^^^^^^^^^^^^^^^^^^
```

and if you look at

http://hackage.haskell.org/package/parsec-3.1.13.0/changelog

the `MonadFail ParsecT` instance was only added in parsec-3.1.12

and it appears that colour-2.3.5 started adding a `MonadFail` constraint to readColourName, see

https://hackage.haskell.org/package/colour-2.3.5/changelog


in other words, as soon as you have `colour-2.3.5`, you need `parsec-3.1.12` (or later) or you'll get that build-failure... 

Does this make any sense? :-)

**Comment by yaxu** [comment]
Thanks @hvr that's really helpful. I've simply added a lower bounds of `3.1.12` to parsec for now.

**Comment by hvr** [comment]
@yaxu btw, this issue can still resurface due to old releases being more liberal with their bounds; I'd suggest tweaking some of the previous `tidal` releases to exclude `colour-2.3.5` -- I can take care of this if you want

**Comment by yaxu** [comment]
You'd be very welcome @hvr, thanks very much!

**Comment by yaxu** [comment]
Hmm now there are different problems..
![image](https://user-images.githubusercontent.com/27940/60248045-bd12ac00-98b9-11e9-8c60-1df7e6e48825.png)

**Comment by yaxu** [comment]
I wonder if it's actually the colour package that needs its dependencies fixing?

**Comment by yaxu** [comment]
Ah no, this just seems to be a 7.8.* version of ghc installed from 2014, which it seems tidal doesn't support any more with its base >=4.8 constraint https://wiki.haskell.org/Base_package

**Comment by hvr** [comment]
@yaxu yes, but if you'd ask cabal to install `tidal` from Hackage, it'd happily backtrack to a recent version that still supported GHC 7.8 (c.f. https://matrix.hackage.haskell.org/#/package/tidal )

**Comment by yaxu** [comment]
Ok @hvr, but I really wouldn't want that to happen - 0.9.5 has a different API so wouldn't work with the rest of the tidal environment (editor plugins and so on)

**Comment by yaxu** [comment]
(Of course it's great that old versions will install, for people who have composed music in the past and want to preserve it)

**Comment by hvr** [comment]
@yaxu ...and that's why you should tell people to somehow record the API they composed their music against...that's what we have version operators for, so you can also say "I need any version of `tidal` compatible with tidal-0.9.5" -- that's what e.g. `tidal ^>= 0.9.5` expresses (which would allow `tidal-0.9.7` but not e.g. `tidal-1.0.10` or `tidal-0.10.1`) :-)
